docs/developer/style.md

```diff
@@ -8,6 +8,153 @@
+
+```golang
+type Listener interface {
+   // Interface definition
+}
+
+// Listener instance for file
+type FileListener struct {
+
+}
+
+// Listener instance for HTTP
+type HttpListener struct {
+
+}
+```
+
+#### Struct initialization
+There are many ways to initialize structs in Golang, base on the use case we can decide which way to initialize objects in Golang.
+In Vald we use [functional option pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) to initialize complex structs. Please read [server.go](https://github.com/vdaas/vald/blob/master/internal/servers/servers.go) and [option.go](https://github.com/vdaas/vald/blob/master/internal/servers/option.go) for the reference implementation.
+The functional options should be separated as another file to improve the readability of the source code, and the method name should be start with `With` word to differentiate with other methods.
+
+Also you can use `&T{}` to initialize the struct. Do not use `new(T)` method to initialize the struct.
+
+### Variables and Constant
+The variable and the constant should be named as:
+- Use MixedCaps
+- Global variable and constant should not use short form unless it is a common terms.
+- Private variable and contant should use short form to improve readability.
+
+The variable and constant name may lead to misunderstanding or confusing, so if the variable and constant name is different to understand, please write some comment even if it is a private member.
+If multiple variables and constants have the same grouping, please use the grouping name as the prefix of the variable and constant name.
+
+Here is some example of the declaration of variables and constants:
+```
+/* Global variables */
+// Same group of variable (error), so add a prefix `Err` to each error variables
+ErrInvalidCacherType = New("invalid cacher type")
+// ErrXXXXXXX
+
+/* Private variables */
+// This variable needs comment in order to understand
+// sds represent the shut down strategy
+sds     []string
+
+// This variable name is common so no comment requires for this variable
+eg      errgroup.Group
+```
+
+### Methods
+The method name should be named as:
+- Use MixedCaps.
+- Should not use short form unless it is a common terms.
+- Should be understandable for everyone even if it is a private method.
+
+#### Getter and Setter
+The Getter and Setter is almost the same as another languages, but the naming convension of the Getter method is different with other languages. Instead of `GetVar1()`, the getter of `Var1` should be the same as the variable name itself `Var1()`.
+
+### Error handling
+
+### Logging
+
+## Program comments
+Program comments makes the code more easier to understand. Basically we suggest not to write many comments inside the source code, unless the source code is very complicated and confusing; otherwise we should divide the source code into methods to keep the readability and usability of the source code.
```

Comment:
```suggestion
Program comments make more easier to understand the source code. Basically we suggest not to write many comments inside the source code unless the source code is very complicated and confusing; otherwise we should divide the source code into methods to keep the readability and usability of the source code.
```